src/main/java/io/divolte/server/IncomingRequestProcessor.java

```diff
@@ -150,14 +154,19 @@ public ProcessingDirective process(final DivolteEvent event) {
 
             if (!duplicate || keepDuplicates) {
                 final GenericRecord avroRecord = mapper.newRecordFromExchange(event);
-                final AvroRecordBuffer avroBuffer = AvroRecordBuffer.fromRecord(
+                try {
+                    final AvroRecordBuffer avroBuffer = AvroRecordBuffer.fromRecord(
                         event.partyCookie,
                         event.sessionCookie,
                         event.requestStartTime,
                         event.clientUtcOffset,
                         avroRecord);
-                listener.incomingRequest(event, avroBuffer, avroRecord);
-                doProcess(avroBuffer);
+                    listener.incomingRequest(event, avroBuffer, avroRecord);
+                    doProcess(avroBuffer);
+                } catch (final RuntimeException e) {
```

Comment:
If you look at the stack trace from #110, you will see it actually propagated up from fromRecord. I guess it wasn't strictly an error in the mapping logic, but rather the generic schema being employed. Either way, I think it is safer to include it.